doc/source/whatsnew/v0.17.0.txt

```diff
@@ -80,6 +80,8 @@ Other enhancements
 
 - ``regex`` argument to ``DataFrame.filter`` now handles numeric column names instead of raising ``ValueError`` (:issue:`10384`).
 
+- ``merge`` will now allow duplicate column names if they are not merged upon (:issue:`10639`).
+
```

Comment:
use ``pd.merge``

if you were a user and saw this, would you want to see a mini-example? (and should we add a mini-example to the merge docs proper?)